### DISCUSSION TITLE
Product Variations

### DISCUSSION THREAD

**Discussion by Alexdimirop** [AshleyRedman/WPClothes2Order#68] [Ideas]
Hi, I just download it and start using it. 

You have done an excellent job of incorporating all these parameters to woocommerce. And I really like that you can see the response in the orders section.

The only difficulty I see is for product variations. 

In each product you enter an SKU with which C2O determines colour, size etc. The problem is that you have to do a separate product for each variation. And this also for the print method. 

And because usually colour,size,print appear on the same page for the client to choose it would be a great option if you can retrieve the SKU from a product variation SKU field (if for example this field is left blank in the plugin).

And so you can make as many variations as you want and simply put different SKU in each variation.

**Comment by AshleyRedman** [comment]
Hey, thanks for the feedback & for checking out the plug-in! 

In regards to product variations, you're completely right and the feature is definitely feasible. 

I will add it onto the roadmap of features to support for the next release (1.1.0), so keep an eye out. Great suggestion by the way!

**Comment by Alexdimirop** [comment]
Great! 

Looking forward to it!

**Comment by ChipHosting** [comment]
Looking forward to Product Variations as I really like this plugin!!! I have found a way around Product Variations for now by using the WooCommerce Composite extension however that comes at a cost I luckily already have a licence for this.

**Reply by AshleyRedman** [reply]
Hi @adamcook102, thank you for the feedback, it's great to hear users of the plugin are finding it useful :) 
I am actively working on the product variations feature. I will be sending out a group email once the feature is live, so make sure you're apart of our mailing list (can be found at the bottom of the page here: https://www.wpclothes2order.com).

**Comment by AshleyRedman** [comment]
Hey @ChipHosting & @Alexdimirop is this feature you are still looking for?

**Reply by Banzarykey** [reply]
Id like to see this feature, adding to that it would be good to update the stock levels in variations. and not sure if this is possible but the have the option to pull the costs of items from C2O feed? maybe with a central config for adding a fixed value or % markup?

**Reply by AshleyRedman** [reply]
@Banzarykey Thanks for the feedback! Variations is still something I am considering as it adds a fair bit of complexity to the plugin. In terms of pulling the costs in etc. I can look into that, C2O may supply that information via their api. When you say to 'pull it in', where would you like that information to be set? As in, if TShirt X costs £10, then I doubt you would want your product to be set to £10, so based on your other comment, setting a % markup, so if the TShirt cost X and yuor markup was 100%, then your product would cost £20?

It's tricky to add in features like so, as others may not use it or expect it to work that way, however, if I can come up with a more generalised solution, I can look at adding that in.
